src/ActionBuilder/Product/SetAttributes.php

```diff
@@ -67,18 +67,22 @@ public function createUpdateActions(
         list($oldAttrs, $variantId) = $variantData;
 
         foreach ($changedValue as $attrIndex => $attr) {
-            $action = $this->startAction($sourceObject, $variantId, $attr, $oldAttrs, $attrIndex);
-
-            // ENUM and LENUM Attributes could contain an empty label (=null) because we only set value and no label.
-            // Therefor, ODM detects a change for the label, which is uninteresting for us.
-            // We don't want any action if the only changed field is the 'null label'.
-            if (array_key_exists('value', $attr) && is_array($attr['value']) && count($attr['value']) === 1) {
-                if (array_key_exists('label', $attr['value']) && $attr['value']['label'] === null) {
-                    continue;
-                }
-            }
+            $attributeName = $attr['name'] ?? $oldAttrs[$attrIndex]['name'];
```

Comment:
Why did you move this here?

Reply:
Because I don't think it is nice to pass three variable to a method just to generate a name.